grails-app/controllers/au/org/ala/userdetails/RegistrationController.groovy

```diff
@@ -150,8 +150,20 @@ class RegistrationController {
 
         if (user) {
             if (params.email != user.email) {
-                // email address has changed, and username and email address must be kept in sync
+                // email address has changed
+                if (userService.isEmailInUse(params.email, user)) {
+                    render(view: "accountError", model: [msg: "Failed to update user profile - A user is already registered with the email address"])
+                    return
+                }
+                // and username and email address must be kept in sync
                 params.userName = params.email
+                // update cookie when email address changed https://github.com/AtlasOfLivingAustralia/userdetails/issues/98
```

Comment:
The cookie should only be updated if `userService.update` succeeds (ie need to move this into the success block below).

Reply:
done. 
added unit tests.